From 060033a3e724c2a081a8b52072b1ababea21e5ca Mon Sep 17 00:00:00 2001
From: Marian Buschsieweke <marian.buschsieweke@ovgu.de>
Date: Thu, 8 Sep 2022 20:36:43 +0200
Subject: [PATCH] src/PosixSerialPort.cpp: add missing include

---
 src/PosixSerialPort.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/PosixSerialPort.cpp b/src/PosixSerialPort.cpp
index 45eaca9..cee31fb 100644
--- a/src/PosixSerialPort.cpp
+++ b/src/PosixSerialPort.cpp
@@ -36,6 +36,7 @@
 #include <termios.h>
 #include <errno.h>
 #include <sys/ioctl.h>
+#include <sys/select.h>
 
 #include <string>
 
-- 
2.37.1

