From 71152ca234e7af2bd96ec3a425976c818edb8753 Mon Sep 17 00:00:00 2001
From: Christian Manal <moenoel@informatik.uni-bremen.de>
Date: Thu, 23 Nov 2017 10:55:07 +0100
Subject: [PATCH 01/12] change header location

---
 core/internals.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/core/internals.h b/core/internals.h
index 30d3942..79c6c4c 100644
--- a/core/internals.h
+++ b/core/internals.h
@@ -61,7 +61,7 @@
 #include <stddef.h>
 #include <stdbool.h>
 
-#include "er-coap-13/er-coap-13.h"
+#include "er-coap-13.h"
 
 #ifdef LWM2M_WITH_LOGS
 #include <inttypes.h>
-- 
2.1.4

