From 85072c812957133c02899e2e1f25dd46f5d576f1 Mon Sep 17 00:00:00 2001
From: Moritz <moritz.holzer@haw-hamburg.de>
Date: Wed, 23 Oct 2024 16:51:12 +0200
Subject: [PATCH] fixed format specifier

---
 examples/client/object_server.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/examples/client/object_server.c b/examples/client/object_server.c
index a6a05bc..a19516d 100644
--- a/examples/client/object_server.c
+++ b/examples/client/object_server.c
@@ -51,6 +51,7 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
+#include <inttypes.h>
 
 typedef struct _server_instance_
 {
@@ -908,7 +909,7 @@ void display_server_object(lwm2m_object_t * object)
     server_instance_t * serverInstance = (server_instance_t *)object->instanceList;
     while (serverInstance != NULL)
     {
-        fprintf(stdout, "    /%u/%u: instanceId: %u, shortServerId: %u, lifetime: %u, storing: %s, binding: %s",
+        fprintf(stdout, "    /%u/%u: instanceId: %u, shortServerId: %u, lifetime: %" PRIu32 ", storing: %s, binding: %s",
                 object->objID, serverInstance->instanceId,
                 serverInstance->instanceId, serverInstance->shortServerId, serverInstance->lifetime,
                 serverInstance->storing ? "true" : "false", serverInstance->binding);
-- 
2.34.1

