From b4060e323b9683f13f6e717ddbbba8fed5bb6d3d Mon Sep 17 00:00:00 2001
From: Christian Manal <moenoel@informatik.uni-bremen.de>
Date: Thu, 23 Nov 2017 11:28:37 +0100
Subject: [PATCH 05/12] fixed warnings in registration.c

---
 core/registration.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/core/registration.c b/core/registration.c
index 55d6ed8..a2fca3f 100644
--- a/core/registration.c
+++ b/core/registration.c
@@ -1283,6 +1283,9 @@ void registration_step(lwm2m_context_t * contextP,
                        time_t currentTime,
                        time_t * timeoutP)
 {
+    (void)contextP;
+    (void)currentTime;
+    (void)timeoutP;
 #ifdef LWM2M_CLIENT_MODE
     lwm2m_server_t * targetP = contextP->serverList;
 
-- 
2.1.4

