From ddbb2dc1078489131c77084934099c8a76c80fb7 Mon Sep 17 00:00:00 2001
From: Christian Manal <moenoel@informatik.uni-bremen.de>
Date: Wed, 29 Nov 2017 16:04:41 +0100
Subject: [PATCH 10/12] fixed warnings registration.c

---
 core/registration.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/core/registration.c b/core/registration.c
index a2fca3f..647a980 100644
--- a/core/registration.c
+++ b/core/registration.c
@@ -69,7 +69,7 @@ static int prv_getRegistrationQueryLength(lwm2m_context_t * contextP,
 {
     int index;
     int res;
-    char buffer[21];
+    uint8_t buffer[21];
 
     index = strlen(QUERY_STARTER QUERY_VERSION_FULL QUERY_DELIMITER QUERY_NAME);
     index += strlen(contextP->endpointName);
@@ -170,7 +170,7 @@ static int prv_getRegistrationQuery(lwm2m_context_t * contextP,
         res = utils_stringCopy(buffer + index, length - index, QUERY_DELIMITER QUERY_LIFETIME);
         if (res < 0) return 0;
         index += res;
-        res = utils_intToText(server->lifetime, buffer + index, length - index);
+        res = utils_intToText(server->lifetime, (uint8_t *)(buffer + index), length - index);
         if (res == 0) return 0;
         index += res;
     }
@@ -535,6 +535,7 @@ static void prv_handleDeregistrationReply(lwm2m_transaction_t * transacP,
                                           void * message)
 {
     lwm2m_server_t * targetP;
+    (void)message;
 
     targetP = (lwm2m_server_t *)(transacP->userData);
     if (NULL != targetP)
-- 
2.1.4

