From 8a20d72f1cb54151995cb30f3650267f896e88d0 Mon Sep 17 00:00:00 2001
From: Christian Manal <moenoel@informatik.uni-bremen.de>
Date: Wed, 29 Nov 2017 16:54:47 +0100
Subject: [PATCH 12/12] fixed warnings in liblwm2m.c

---
 core/liblwm2m.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/core/liblwm2m.c b/core/liblwm2m.c
index bf4b1b1..b7aafd7 100644
--- a/core/liblwm2m.c
+++ b/core/liblwm2m.c
@@ -365,7 +365,9 @@ int lwm2m_step(lwm2m_context_t * contextP,
                time_t * timeoutP)
 {
     time_t tv_sec;
+#ifdef LWM2M_CLIENT_MODE
     int result;
+#endif
 
     LOG_ARG("timeoutP: %" PRId64, *timeoutP);
     tv_sec = lwm2m_gettime();
-- 
2.1.4

