From c3d723c0544b51dcea92453856447a4b9da6d288 Mon Sep 17 00:00:00 2001
From: Leandro Lanzieri <leandro.lanzieri@haw-hamburg.de>
Date: Mon, 4 Feb 2019 12:02:14 +0100
Subject: [PATCH] change liblwm2m.h include in er-coap-13.c

---
 core/er-coap-13/er-coap-13.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/core/er-coap-13/er-coap-13.c b/core/er-coap-13/er-coap-13.c
index 7b9d43a..647bb88 100644
--- a/core/er-coap-13/er-coap-13.c
+++ b/core/er-coap-13/er-coap-13.c
@@ -46,7 +46,7 @@
 
 #include "er-coap-13.h"
 
-#include "../liblwm2m.h" /* for lwm2m_malloc() and lwm2m_free() */
+#include "liblwm2m.h" /* for lwm2m_malloc() and lwm2m_free() */
 
 #define DEBUG 0
 #if DEBUG
-- 
2.17.1
