From e3fcae282ad8c1788fd082dc5431cc71944e64da Mon Sep 17 00:00:00 2001
From: Leandro Lanzieri <leandro.lanzieri@haw-hamburg.de>
Date: Tue, 19 Feb 2019 17:36:50 +0100
Subject: [PATCH] fix access control object read

---
 examples/client/object_access_control.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/examples/client/object_access_control.c b/examples/client/object_access_control.c
index e2b1f77..149ec24 100644
--- a/examples/client/object_access_control.c
+++ b/examples/client/object_access_control.c
@@ -95,7 +95,7 @@ static uint8_t prv_set_tlv(lwm2m_data_t* dataP, acc_ctrl_oi_t* accCtrlOiP)
                 subTlvP[ri].id = accCtrlRiP->resInstId;
                 lwm2m_data_encode_int(accCtrlRiP->accCtrlValue, &subTlvP[ri]);
             }
-            lwm2m_data_encode_instances(subTlvP, 2, dataP);
+            lwm2m_data_encode_instances(subTlvP, ri, dataP);
             return COAP_205_CONTENT;
         }
     }   break;
-- 
2.17.1
