From 4cf9ceba3052b385eb2378e474307e1eb4b90aac Mon Sep 17 00:00:00 2001
From: Leandro Lanzieri <leandro.lanzieri@haw-hamburg.de>
Date: Thu, 1 Feb 2024 10:23:27 +0100
Subject: [PATCH 1/1] Add const modifier to URI on lwm2m_resource_value_changed

---
 core/liblwm2m.h | 2 +-
 core/observe.c  | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/core/liblwm2m.h b/core/liblwm2m.h
index 15fac32..4a439b7 100644
--- a/core/liblwm2m.h
+++ b/core/liblwm2m.h
@@ -696,7 +696,7 @@ int lwm2m_remove_object(lwm2m_context_t * contextP, uint16_t id);
 // If withObjects is true, the registration update contains the object list.
 int lwm2m_update_registration(lwm2m_context_t * contextP, uint16_t shortServerID, bool withObjects);
 
-void lwm2m_resource_value_changed(lwm2m_context_t * contextP, lwm2m_uri_t * uriP);
+void lwm2m_resource_value_changed(lwm2m_context_t * contextP, const lwm2m_uri_t * uriP);
 #endif
 
 #ifdef LWM2M_SERVER_MODE
diff --git a/core/observe.c b/core/observe.c
index 308448d..3494799 100644
--- a/core/observe.c
+++ b/core/observe.c
@@ -445,7 +445,7 @@ lwm2m_observed_t * observe_findByUri(lwm2m_context_t * contextP,
 }
 
 void lwm2m_resource_value_changed(lwm2m_context_t * contextP,
-                                  lwm2m_uri_t * uriP)
+                                  const lwm2m_uri_t * uriP)
 {
     lwm2m_observed_t * targetP;
 
-- 
2.43.0

