From 2304bef6c63b15411fae3050554edc0538ae93d9 Mon Sep 17 00:00:00 2001
From: Francisco Molina <femolina@uc.cl>
Date: Fri, 14 Aug 2020 15:08:22 +0200
Subject: [PATCH 4/7] sys/uwbcfg: use DPL_ENOENT instead of OS_ENOENT

---
 sys/uwbcfg/src/uwbcfg.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/sys/uwbcfg/src/uwbcfg.c b/sys/uwbcfg/src/uwbcfg.c
index 595c19b..4c4678e 100644
--- a/sys/uwbcfg/src/uwbcfg.c
+++ b/sys/uwbcfg/src/uwbcfg.c
@@ -149,7 +149,7 @@ uwbcfg_set(int argc, char **argv, char *val)
                 return CONF_VALUE_SET(val, CONF_STRING, g_uwb_config[i]);
         }
     }
-    return OS_ENOENT;
+    return DPL_ENOENT;
 }
 
 char*
-- 
2.28.0

