From 3409c3834022c9e7cd73ac69e34dfaa101947cf0 Mon Sep 17 00:00:00 2001
From: Oleg Hahm <oleg@hobbykeller.org>
Date: Tue, 4 Aug 2015 23:09:27 +0200
Subject: [PATCH 8/8] fix debug out in PDU parsing

---
 net.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/net.c b/net.c
index 9372570..86c7793 100644
--- a/net.c
+++ b/net.c
@@ -930,9 +930,10 @@ if (!coap_pdu_parse((unsigned char *)buf, bytes_read, node->pdu)) {
 #endif
     unsigned char addr[INET6_ADDRSTRLEN+8];

-    if (coap_print_addr(&src, addr, INET6_ADDRSTRLEN+8))
+    if (coap_print_addr(&src, addr, INET6_ADDRSTRLEN+8)) {
       debug("** received %d bytes from %s:\n", (int)bytes_read, addr);

+    }
     coap_show_pdu( node->pdu );
   }
 #endif
--
2.5.0
