From d48f439824ecc17b3fd9cc1d9b8cbea7b00a5b0c Mon Sep 17 00:00:00 2001
From: tobhe <t.heider@campus.lmu.de>
Date: Wed, 19 Jul 2017 13:11:05 +0200
Subject: [PATCH 1/2] Fixed fallthrough compiler warning

---
 heatshrink_encoder.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/heatshrink_encoder.c b/heatshrink_encoder.c
index edf4abe..602e177 100644
--- a/heatshrink_encoder.c
+++ b/heatshrink_encoder.c
@@ -235,6 +235,7 @@ HSE_poll_res heatshrink_encoder_poll(heatshrink_encoder *hse,
             break;
         case HSES_FLUSH_BITS:
             hse->state = st_flush_bit_buffer(hse, &oi);
+            return HSER_POLL_EMPTY;
         case HSES_DONE:
             return HSER_POLL_EMPTY;
         default:
-- 
2.13.3

