From b21ceb6f079a63d05552831d372de5d8bbda1173 Mon Sep 17 00:00:00 2001
From: Gunar Schorcht <gunar@schorcht.net>
Date: Mon, 20 Jan 2025 21:49:10 +0100
Subject: [PATCH 01/28] compilation: avoid multiple definitions of macros

---
 components/xtensa/include/xtensa/xtruntime-frames.h | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/components/xtensa/include/xtensa/xtruntime-frames.h b/components/xtensa/include/xtensa/xtruntime-frames.h
index b78095e0dd..4a5b4d356a 100644
--- a/components/xtensa/include/xtensa/xtruntime-frames.h
+++ b/components/xtensa/include/xtensa/xtruntime-frames.h
@@ -29,6 +29,13 @@
 
 #include <xtensa/config/core.h>
 
+#ifdef STRUCT_BEGIN
+#undef STRUCT_BEGIN
+#undef STRUCT_FIELD
+#undef STRUCT_AFIELD
+#undef STRUCT_END
+#endif
+
 /*  Macros that help define structures for both C and assembler:  */
 #if defined(_ASMLANGUAGE) || defined(__ASSEMBLER__)
 
-- 
2.34.1

