From f935f3912142ca013e0dc9c6e14aab3d16664874 Mon Sep 17 00:00:00 2001
From: Gunar Schorcht <gunar@schorcht.net>
Date: Sun, 30 Jan 2022 17:01:16 +0100
Subject: [PATCH 05/12] wpa_supplicant: do not require mbedtls

---
 components/wpa_supplicant/port/os_xtensa.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/components/wpa_supplicant/port/os_xtensa.c b/components/wpa_supplicant/port/os_xtensa.c
index cf92aeb5e4..a384a97d2e 100644
--- a/components/wpa_supplicant/port/os_xtensa.c
+++ b/components/wpa_supplicant/port/os_xtensa.c
@@ -29,7 +29,9 @@
 #include <sys/time.h>
 #include "esp_system.h"
 #include "utils/common.h"
+#ifdef USE_MBEDTLS_CRYPTO
 #include "mbedtls/platform_util.h"
+#endif
 
 int os_get_time(struct os_time *t)
 {
-- 
2.17.1

