From b63a8db9ecc824fd05ab19b51fbc7c217234e582 Mon Sep 17 00:00:00 2001
From: Gunar Schorcht <gunar@schorcht.net>
Date: Sat, 1 Mar 2025 17:40:10 +0100
Subject: [PATCH 07/28] sdmmc: avoid type definition conflicts

---
 components/sdmmc/include/sd_protocol_types.h | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/components/sdmmc/include/sd_protocol_types.h b/components/sdmmc/include/sd_protocol_types.h
index 70cdf76d12..41fac5d339 100644
--- a/components/sdmmc/include/sd_protocol_types.h
+++ b/components/sdmmc/include/sd_protocol_types.h
@@ -34,6 +34,7 @@
 extern "C" {
 #endif
 
+#if !defined(RIOT_VERSION)
 /**
  * Decoded values from SD card Card Specific Data register
  */
@@ -96,6 +97,8 @@ typedef struct {
     uint8_t sec_feature;        /*!< secure data management features supported by the card */
 } sdmmc_ext_csd_t;
 
+#endif /* !defined(RIOT_VERSION) */
+
 /**
  * SD/MMC command response buffer
  */
@@ -245,6 +248,7 @@ typedef struct {
     esp_err_t (*is_slot_set_to_uhs1)(int slot, bool *is_uhs1); /*!< host slot is set to uhs1 or not*/
 } sdmmc_host_t;
 
+#if !defined(RIOT_VERSION)
 /**
  * SD/MMC card information structure
  */
@@ -273,6 +277,8 @@ typedef struct {
     uint32_t reserved : 22;     /*!< Reserved for future expansion */
 } sdmmc_card_t;
 
+#endif /* !defined(RIOT_VERSION) */
+
 /**
  * SD/MMC erase command(38) arguments
  * SD:
-- 
2.34.1

