From e24009b2ff743a90d123816280437ecc0a755847 Mon Sep 17 00:00:00 2001
From: Gunar Schorcht <gunar@schorcht.net>
Date: Sun, 30 Jan 2022 08:17:34 +0100
Subject: [PATCH 08/12] esp_log: use RIOT version of log module

---
 components/log/include/esp_log.h | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/components/log/include/esp_log.h b/components/log/include/esp_log.h
index 0e3148f62d..5419a623e2 100644
--- a/components/log/include/esp_log.h
+++ b/components/log/include/esp_log.h
@@ -155,6 +155,8 @@ void esp_log_write(esp_log_level_t level, const char* tag, const char* format, .
  */
 void esp_log_writev(esp_log_level_t level, const char* tag, const char* format, va_list args);
 
+#if !defined(RIOT_VERSION)
+
 /** @cond */
 
 #include "esp_log_internal.h"
@@ -498,6 +500,8 @@ void esp_log_writev(esp_log_level_t level, const char* tag, const char* format,
 #endif // !(defined(__cplusplus) && (__cplusplus >  201703L))
 /** @endcond */
 
+#endif /* !defined(RIOT_VERSION) */
+
 #ifdef __cplusplus
 }
 #endif
-- 
2.17.1

