From 321ef67c4675f566dcbe885558ece59e3a4f3845 Mon Sep 17 00:00:00 2001
From: Gunar Schorcht <gunar@schorcht.net>
Date: Sat, 1 Mar 2025 17:56:30 +0100
Subject: [PATCH 09/28] bt/controller: fix printf format string

---
 components/bt/controller/esp32/bt.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/components/bt/controller/esp32/bt.c b/components/bt/controller/esp32/bt.c
index d94c9c0507..6a20c2b1b7 100644
--- a/components/bt/controller/esp32/bt.c
+++ b/components/bt/controller/esp32/bt.c
@@ -1273,7 +1273,7 @@ static void btdm_controller_mem_init(void)
 {
     /* initialise .data section */
     memcpy(&_data_start_btdm, (void *)_data_start_btdm_rom, &_data_end_btdm - &_data_start_btdm);
-    ESP_LOGD(BTDM_LOG_TAG, ".data initialise [0x%08x] <== [0x%08x]", (uint32_t)&_data_start_btdm, _data_start_btdm_rom);
+    ESP_LOGD(BTDM_LOG_TAG, ".data initialise [0x%08"PRIx32"] <== [0x%08"PRIx32"]", (uint32_t)&_data_start_btdm, _data_start_btdm_rom);
 
     //initial em, .bss section
     for (int i = 1; i < sizeof(btdm_dram_available_region)/sizeof(btdm_dram_available_region_t); i++) {
-- 
2.34.1

