From c6789a3f627915ff7d9f6909c2202eac2cb9db0c Mon Sep 17 00:00:00 2001
From: Gunar Schorcht <gunar@schorcht.net>
Date: Tue, 31 May 2022 22:32:47 +0200
Subject: [PATCH 23/23] bootloader: remove compile time from banner

---
 components/bootloader_support/src/bootloader_init.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/components/bootloader_support/src/bootloader_init.c b/components/bootloader_support/src/bootloader_init.c
index 1d73f5dc26..9e45b1d415 100644
--- a/components/bootloader_support/src/bootloader_init.c
+++ b/components/bootloader_support/src/bootloader_init.c
@@ -91,5 +91,7 @@ void bootloader_enable_random(void)
 void bootloader_print_banner(void)
 {
     ESP_LOGI(TAG, "ESP-IDF %s 2nd stage bootloader", IDF_VER);
+#if 0 /* not used in RIOT */
     ESP_LOGI(TAG, "compile time " __TIME__);
+#endif
 }
-- 
2.17.1

