From 06c5e3e1804a06e14d3965beaa1b42f7a3ff06e0 Mon Sep 17 00:00:00 2001
From: Gunar Schorcht <gunar@schorcht.net>
Date: Mon, 7 Apr 2025 14:54:49 +0200
Subject: [PATCH 34/34] ieee802154/driver: fix misssing include

---
 components/ieee802154/driver/esp_ieee802154_dev.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/components/ieee802154/driver/esp_ieee802154_dev.c b/components/ieee802154/driver/esp_ieee802154_dev.c
index d7a1b4f269..a38957b4a0 100644
--- a/components/ieee802154/driver/esp_ieee802154_dev.c
+++ b/components/ieee802154/driver/esp_ieee802154_dev.c
@@ -14,6 +14,7 @@
 #include "esp_check.h"
 #include "esp_coex_i154.h"
 #include "esp_err.h"
+#include "esp_intr_types.h"
 #include "esp_log.h"
 #include "esp_timer.h"
 #include "esp_ieee802154_ack.h"
-- 
2.34.1

