From 21202804f26b194607a412476a96f03d3df30688 Mon Sep 17 00:00:00 2001
From: Hinnerk van Bruinehsen <h.v.bruinehsen@fu-berlin.de>
Date: Wed, 29 Oct 2014 12:11:29 +0100
Subject: [PATCH 9/9] Add missing include

---
 src-api/rfc5444/rfc5444_tlv_writer.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src-api/rfc5444/rfc5444_tlv_writer.h b/src-api/rfc5444/rfc5444_tlv_writer.h
index ace7313..8d0ce3a 100644
--- a/src-api/rfc5444/rfc5444_tlv_writer.h
+++ b/src-api/rfc5444/rfc5444_tlv_writer.h
@@ -43,6 +43,7 @@
 #define RFC5444_TLV_WRITER_H_
 
 #include "common/common_types.h"
+#include "rfc5444_context.h"
 
 struct rfc5444_tlv_writer_data {
   uint8_t *buffer;
-- 
2.1.2

