From ca34f5c4e808b9b6e61e1ceba5e7065bf71fe37a Mon Sep 17 00:00:00 2001
From: Jana Eisoldt <jana.eisoldt@st.ovgu.de>
Date: Mon, 21 Feb 2022 16:52:01 +0100
Subject: [PATCH] parse_elffile: fix riot_base

---
 cosy.py | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/cosy.py b/cosy.py
index b36c28a..2081f64 100755
--- a/cosy.py
+++ b/cosy.py
@@ -154,9 +154,9 @@ def parse_elffile(elffile, prefix, appdir, riot_base=None):
     rbase = ["riotbuild/riotproject"]
     if riot_base:
         rbase.append(riot_base.strip("/"))
-    else:
-        rbase.append("RIOT")
-        rbase.append("riotbuild/riotbase")
+
+    rbase.append("RIOT")
+    rbase.append("riotbuild/riotbase")
     riot_base = "|".join([f'{p}/build|{p}' for p in rbase])
 
     c = re.compile(r"(?P<addr>[0-9a-f]+) "
