From 3f675f891794b3655c06b721c7f34c462233e7a5 Mon Sep 17 00:00:00 2001
From: Martine Lenders <mail@martine-lenders.eu>
Date: Fri, 26 Aug 2016 11:48:26 +0200
Subject: [PATCH] RIOT adapter: port to new netreg struct

---
 src/ccn-lite-riot.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/ccn-lite-riot.c b/src/ccn-lite-riot.c
index b5e7660..ba0a2cc 100644
--- a/src/ccn-lite-riot.c
+++ b/src/ccn-lite-riot.c
@@ -229,8 +229,8 @@ ccnl_open_netif(kernel_pid_t if_pid, gnrc_nettype_t netreg_type)
     /* configure the interface to use the specified nettype protocol */
     gnrc_netapi_set(if_pid, NETOPT_PROTO, 0, &netreg_type, sizeof(gnrc_nettype_t));
     /* register for this nettype */
-    _ccnl_ne.demux_ctx =  GNRC_NETREG_DEMUX_CTX_ALL;
-    _ccnl_ne.pid = _ccnl_event_loop_pid;
+    gnrc_netreg_entry_init_pid(&_ccnl_ne, GNRC_NETREG_DEMUX_CTX_ALL,
+                               _ccnl_event_loop_pid);
     return gnrc_netreg_register(netreg_type, &_ccnl_ne);
 }
 
-- 
2.7.4

