From 5aac38a737d95995b6fa5655efb8d43bfa819a62 Mon Sep 17 00:00:00 2001
From: Martine Lenders <mail@martine-lenders.eu>
Date: Thu, 27 Jul 2017 13:17:35 +0200
Subject: [PATCH] ccn-line-riot: port to new gnrc_netif

---
 src/ccn-lite-riot.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/ccn-lite-riot.c b/src/ccn-lite-riot.c
index 3b8db4d..c9eec2f 100644
--- a/src/ccn-lite-riot.c
+++ b/src/ccn-lite-riot.c
@@ -228,7 +228,7 @@ int
 ccnl_open_netif(kernel_pid_t if_pid, gnrc_nettype_t netreg_type)
 {
     assert(pid_is_valid(if_pid));
-    if (!gnrc_netif_exist(if_pid)) {
+    if (gnrc_netif_get_by_pid(if_pid) == NULL) {
         return -1;
     }
     if (ccnl_relay.ifcount >= CCNL_MAX_INTERFACES) {
-- 
2.7.4

