From 34075e9fb91f5e4531d39655c616a7013b2e6dfa Mon Sep 17 00:00:00 2001
From: Benjamin Valentin <benpicco@googlemail.com>
Date: Wed, 6 May 2020 16:56:46 +0200
Subject: [PATCH] esptool.py: use python3

---
 esptool.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/esptool.py b/esptool.py
index aa10272..b2acd57 100755
--- a/esptool.py
+++ b/esptool.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 #
 # ESP8266 & ESP32 family ROM Bootloader Utility
 # Copyright (C) 2014-2016 Fredrik Ahlberg, Angus Gratton, Espressif Systems (Shanghai) PTE LTD, other contributors as noted.
-- 
2.25.1

