--- ./bin/pkg/native/fatfs/ffconf.h	2017-05-08 12:19:04.813286471 +0200
+++ ./bin/pkg/native/fatfs/ffconf_patched.h	2017-05-08 12:19:28.785574968 +0200
@@ -202,7 +202,7 @@
 / System Configurations
 /---------------------------------------------------------------------------*/
 
-#define	_FS_TINY	0
+#define	_FS_TINY	1
 /* This option switches tiny buffer configuration. (0:Normal or 1:Tiny)
 /  At the tiny configuration, size of file object (FIL) is reduced _MAX_SS bytes.
 /  Instead of private sector buffer eliminated from the file object, common sector
