From 67549da2f1f5e22e7789443f23c6a8bb470bc3f9 Mon Sep 17 00:00:00 2001
From: Benjamin Valentin <benpicco@zedat.fu-berlin.de>
Date: Fri, 10 Oct 2014 02:08:32 +0200
Subject: [PATCH 06/10] if_index is not used

---
 src-api/common/netaddr.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src-api/common/netaddr.c b/src-api/common/netaddr.c
index 6b214ee..0b8fc1d 100644
--- a/src-api/common/netaddr.c
+++ b/src-api/common/netaddr.c
@@ -317,6 +317,7 @@ netaddr_create_host_bin(struct netaddr *host, const struct netaddr *netmask,
 int
 netaddr_socket_init(union netaddr_socket *combined, const struct netaddr *addr,
     uint16_t port, unsigned if_index) {
+  (void)if_index;
   /* initialize memory block */
   memset(combined, 0, sizeof(*combined));

--
1.9.1
