From cc4fa04e6240d7a63c021a6c5d3d8a010fbd888c Mon Sep 17 00:00:00 2001
From: Brenton Chetty <brent7984@gmail.com>
Date: Wed, 21 Aug 2019 10:21:00 +0200
Subject: [PATCH] Adjust net.c for gcc-arm compiler

---
 src/unix/net.c | 15 +++------------
 1 file changed, 3 insertions(+), 12 deletions(-)

diff --git a/src/unix/net.c b/src/unix/net.c
index 17535b5..93de339 100644
--- a/src/unix/net.c
+++ b/src/unix/net.c
@@ -13,7 +13,7 @@
 
 #include "zenoh.h"
 #include "zenoh/net.h"
-
+#define MSG_NOSIGNAL	0x4000
 z_socket_result_t
 open_tx_session(const char *locator) {
   z_socket_result_t r;
@@ -106,18 +106,9 @@ int z_compute_remaining(struct iovec* iov, int iovcnt, size_t sent) {
 
 int z_send_iovec(z_socket_t sock, struct iovec* iov, int iovcnt) {
   int len = 0;
-  for (int i = 0; i < iovcnt; ++i) 
+  for (int i = 0; i < iovcnt; ++i){ 
     len += iov[i].iov_len;
-  
-  int n = writev(sock, iov, iovcnt);
-  Z_DEBUG_VA("z_send_iovec sent %d of %d bytes \n", n, len);
-  while (n < len) {
-    iovcnt = z_compute_remaining(iov, iovcnt, n);
-    len = z_iovs_len(iov, iovcnt);    
-    n = writev(sock, iov, iovcnt);
-    Z_DEBUG_VA("z_send_iovec sent %d of %d bytes \n", n, len);
-    if (n < 0) 
-      return -1;    
+	write(sock, iov[i].iov_base, iov[i].iov_len);
   }
   return 0;
 }
-- 
2.20.1

