From 23eff8ffe148e23931c6da226eb207185a55879b Mon Sep 17 00:00:00 2001
From: Benjamin Valentin <benjamin.valentin@ml-pa.com>
Date: Wed, 25 Aug 2021 16:19:00 +0200
Subject: [PATCH] stm32f3xx: remove ErrorStatus

---
 Include/stm32f3xx.h | 6 ------
 1 file changed, 6 deletions(-)

diff --git a/Include/stm32f3xx.h b/Include/stm32f3xx.h
index 447e6c4..52306f4 100644
--- a/Include/stm32f3xx.h
+++ b/Include/stm32f3xx.h
@@ -174,12 +174,6 @@ typedef enum
 } FunctionalState;
 #define IS_FUNCTIONAL_STATE(STATE) (((STATE) == DISABLE) || ((STATE) == ENABLE))
 
-typedef enum 
-{
-  SUCCESS = 0U,
-  ERROR = !SUCCESS
-} ErrorStatus;
-
 /**
   * @}
   */
-- 
2.30.2

