From d35146dc1cf3c766f3d9bc6c8e83ef438047fb42 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Mikolai=20G=C3=BCtschow?= <mikolai.guetschow@tu-dresden.de>
Date: Thu, 29 Aug 2024 11:00:07 +0200
Subject: [PATCH] emlearn/eml_common.h: remove redundant check

---
 emlearn/eml_common.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/emlearn/eml_common.h b/emlearn/eml_common.h
index e4d2adf..1f8a599 100644
--- a/emlearn/eml_common.h
+++ b/emlearn/eml_common.h
@@ -37,7 +37,7 @@ eml_error_strs[EmlErrors] = {
 
 bool
 eml_error_valid(EmlError e) {
-    return (e >= EmlOk && e < EmlErrors);
+    return (e < EmlErrors);
 }
 
 /**
-- 
2.39.2

