From 1efd759af2cdb13291bba836bcf2775b1dea8924 Mon Sep 17 00:00:00 2001
From: Juan Carrano <j.carrano@fu-berlin.de>
Date: Mon, 7 May 2018 13:29:15 +0200
Subject: [PATCH 3/7] Make size of LoadF buffer configurable.

---
 lauxlib.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lauxlib.c b/lauxlib.c
index 018a4347..8bd2b5af 100644
--- a/lauxlib.c
+++ b/lauxlib.c
@@ -635,7 +635,7 @@ LUALIB_API void luaL_unref (lua_State *L, int t, int ref) {
 typedef struct LoadF {
   int n;  /* number of pre-read characters */
   FILE *f;  /* file being read */
-  char buff[BUFSIZ];  /* area for reading file */
+  char buff[LUAL_BUFFERSIZE];  /* area for reading file */
 } LoadF;
 
 
-- 
2.25.1

