*** stock_iot-lab_M3/openwsn/02a-MAClow/topology.c	Thu Apr 24 11:01:36 2014
--- riot-openwsn-wip/openwsn/02a-MAClow/topology.c	Thu Apr 24 16:55:54 2014
***************
*** 4,13 ****

  //=========================== defines =========================================

! #define TOPOLOGY_MOTE1 0x6f
! #define TOPOLOGY_MOTE2 0xb9
! #define TOPOLOGY_MOTE3 0x3b
!
  //=========================== variables =======================================

  //=========================== prototypes ======================================
--- 4,13 ----

  //=========================== defines =========================================

! #define TOPOLOGY_MOTE1 0x01
! #define TOPOLOGY_MOTE2 0x02
! #define TOPOLOGY_MOTE3 0x03
! #define TOPOLOGY_MOTE4 0x04
  //=========================== variables =======================================

  //=========================== prototypes ======================================
***************
*** 32,39 ****
              returnVal=FALSE;
           }
           break;
!       case TOPOLOGY_MOTE3:
!          if (ieee802514_header->src.addr_64b[7]==TOPOLOGY_MOTE2) {
              returnVal=TRUE;
           } else {
              returnVal=FALSE;
--- 32,47 ----
              returnVal=FALSE;
           }
           break;
!        case TOPOLOGY_MOTE3:
!          if (ieee802514_header->src.addr_64b[7]==TOPOLOGY_MOTE2 ||
!              ieee802514_header->src.addr_64b[7]==TOPOLOGY_MOTE4) {
!             returnVal=TRUE;
!          } else {
!             returnVal=FALSE;
!          }
!          break;
!        case TOPOLOGY_MOTE4:
!          if (ieee802514_header->src.addr_64b[7]==TOPOLOGY_MOTE3) {
              returnVal=TRUE;
           } else {
              returnVal=FALSE;
