*** stock_iot-lab_M3/openwsn/03b-IPv6/icmpv6.c	Thu Apr 24 11:01:36 2014
--- riot-openwsn-wip/openwsn/03b-IPv6/icmpv6.c	Thu Apr 24 16:55:54 2014
***************
*** 12,27 ****

  //=========================== public ==========================================

! void icmpv6_init() {
  }

! error_t icmpv6_send(OpenQueueEntry_t* msg) {
     msg->owner       = COMPONENT_ICMPv6;
     msg->l4_protocol = IANA_ICMPv6;
     return forwarding_send(msg);
  }

! void icmpv6_sendDone(OpenQueueEntry_t* msg, error_t error) {
     msg->owner = COMPONENT_ICMPv6;
     switch (msg->l4_sourcePortORicmpv6Type) {
        case IANA_ICMPv6_ECHO_REQUEST:
--- 12,27 ----

  //=========================== public ==========================================

! void icmpv6_init(void) {
  }

! owerror_t icmpv6_send(OpenQueueEntry_t* msg) {
     msg->owner       = COMPONENT_ICMPv6;
     msg->l4_protocol = IANA_ICMPv6;
     return forwarding_send(msg);
  }

! void icmpv6_sendDone(OpenQueueEntry_t* msg, owerror_t error) {
     msg->owner = COMPONENT_ICMPv6;
     switch (msg->l4_sourcePortORicmpv6Type) {
        case IANA_ICMPv6_ECHO_REQUEST:
