*** stock_iot-lab_M3/openwsn/03b-IPv6/forwarding.h	Thu Apr 24 11:01:36 2014
--- riot-openwsn-wip/openwsn/03b-IPv6/forwarding.h	Thu Apr 24 16:55:54 2014
***************
*** 10,22 ****
--- 10,32 ----

  #include "iphc.h"

+
  //=========================== define ==========================================

+ #define RPL_HOPBYHOP_HEADER_OPTION_TYPE  0x63
+
  enum {
     PCKTFORWARD     = 1,
     PCKTSEND        = 2,
  };

+ enum {
+   O_FLAG   = 0x80,
+   R_FLAG   = 0x40,
+   F_FLAG   = 0x20,
+ };
+
+
  //=========================== typedef =========================================

  /**
***************
*** 40,49 ****

  //=========================== prototypes ======================================

! void    forwarding_init();
! error_t forwarding_send(OpenQueueEntry_t *msg);
! void    forwarding_sendDone(OpenQueueEntry_t* msg, error_t error);
! void    forwarding_receive(OpenQueueEntry_t* msg, ipv6_header_iht ipv6_header);

  /**
  \}
--- 50,62 ----

  //=========================== prototypes ======================================

! void    forwarding_init(void);
! owerror_t forwarding_send(OpenQueueEntry_t *msg);
! void    forwarding_sendDone(OpenQueueEntry_t *msg, owerror_t error);
! void    forwarding_receive(OpenQueueEntry_t *msg,
!                            ipv6_header_iht ipv6_header,
!                            ipv6_hopbyhop_ht ipv6_hop_header,
!                            rpl_hopoption_ht hop_by_hop_option);

  /**
  \}
