*** stock_iot-lab_M3/openwsn/04-TRAN/openudp.h	Thu Apr 24 11:01:36 2014
--- riot-openwsn-wip/openwsn/04-TRAN/openudp.h	Thu Apr 24 16:55:54 2014
***************
*** 4,10 ****
  /**
  \addtogroup Transport
  \{
! \addtogroup UDP
  \{
  */

--- 4,10 ----
  /**
  \addtogroup Transport
  \{
! \addtogroup OpenUdp
  \{
  */

***************
*** 45,55 ****

  //=========================== prototypes ======================================

! void    openudp_init();
! error_t openudp_send(OpenQueueEntry_t* msg);
! void    openudp_sendDone(OpenQueueEntry_t* msg, error_t error);
  void    openudp_receive(OpenQueueEntry_t* msg);
! bool    openudp_debugPrint();

  /**
  \}
--- 45,55 ----

  //=========================== prototypes ======================================

! void    openudp_init(void);
! owerror_t openudp_send(OpenQueueEntry_t* msg);
! void    openudp_sendDone(OpenQueueEntry_t* msg, owerror_t error);
  void    openudp_receive(OpenQueueEntry_t* msg);
! bool    openudp_debugPrint(void);

  /**
  \}
