*** stock_iot-lab_M3/openwsn/04-TRAN/rsvp.h	Thu Apr 24 11:01:36 2014
--- riot-openwsn-wip/openwsn/04-TRAN/rsvp.h	Thu Apr 24 16:55:54 2014
***************
*** 157,163 ****



! void    rsvp_init();
  void    rsvp_qos_request(uint8_t bandwith, uint16_t refresh_period,open_addr_t dest);

  #endif
--- 157,163 ----



! void    rsvp_init(void);
  void    rsvp_qos_request(uint8_t bandwith, uint16_t refresh_period,open_addr_t dest);

  #endif
