*** stock_iot-lab_M3/openwsn/04-TRAN/rsvp.c	Thu Apr 24 11:01:36 2014
--- riot-openwsn-wip/openwsn/04-TRAN/rsvp.c	Thu Apr 24 16:55:54 2014
***************
*** 17,27 ****
     uint8_t rsvp_timer_id;
  }rsvp_vars_t;

! void rsvp_timer_cb();

  rsvp_vars_t rsvp_vars;

! void rsvp_init(){
     rsvp_vars.rsvp_period    = 0;
     rsvp_vars.rsvp_timer_id  = 0;
  }
--- 17,27 ----
     uint8_t rsvp_timer_id;
  }rsvp_vars_t;

! void rsvp_timer_cb(void);

  rsvp_vars_t rsvp_vars;

! void rsvp_init(void){
     rsvp_vars.rsvp_period    = 0;
     rsvp_vars.rsvp_timer_id  = 0;
  }
***************
*** 35,42 ****
  void rsvp_qos_request(uint8_t bandwith, uint16_t refresh_period, open_addr_t dest){

        OpenQueueEntry_t* pkt;
!       error_t           outcome;
        uint8_t           i,j;

        pkt = openqueue_getFreePacketBuffer(COMPONENT_RSVP);
        if (pkt==NULL) {
--- 35,46 ----
  void rsvp_qos_request(uint8_t bandwith, uint16_t refresh_period, open_addr_t dest){

        OpenQueueEntry_t* pkt;
!       owerror_t           outcome;
        uint8_t           i,j;
+
+       (void)outcome;
+       (void)i;
+       (void)j;

        pkt = openqueue_getFreePacketBuffer(COMPONENT_RSVP);
        if (pkt==NULL) {
