*** stock_iot-lab_M3/openwsn/07-App/udpstorm/udpstorm.h	Thu Apr 24 11:01:37 2014
--- riot-openwsn-wip/openwsn/07-App/udpstorm/udpstorm.h	Thu Apr 24 16:53:30 2014
***************
*** 2,8 ****
  #define __UDPSTORM_H

  /**
! \addtogroup App
  \{
  \addtogroup udpStorm
  \{
--- 2,8 ----
  #define __UDPSTORM_H

  /**
! \addtogroup AppUdp
  \{
  \addtogroup udpStorm
  \{
***************
*** 16,22 ****

  //=========================== prototypes ======================================

! void udpstorm_init();

  /**
  \}
--- 16,22 ----

  //=========================== prototypes ======================================

! void udpstorm_init(void);

  /**
  \}
