*** stock_iot-lab_M3/openwsn/07-App/ohlone/ohlone.h	Thu Apr 24 11:01:37 2014
--- riot-openwsn-wip/openwsn/07-App/ohlone/ohlone.h	Thu Apr 24 16:53:29 2014
***************
*** 2,27 ****
  #define __OHLONE_H

  /**
! \addtogroup App
  \{
  \addtogroup ohlone
  \{
  */

  //=========================== define ==========================================

  //=========================== typedef =========================================

! //=========================== variables =======================================

  //=========================== prototypes ======================================

! void ohlone_init();
! bool ohlone_shouldIlisten();
  void ohlone_receive(OpenQueueEntry_t* msg);
! void ohlone_sendDone(OpenQueueEntry_t* msg, error_t error);
! void ohlone_connectDone(error_t error);
! bool ohlone_debugPrint();

  /**
  \}
--- 2,36 ----
  #define __OHLONE_H

  /**
! \addtogroup AppTcp
  \{
  \addtogroup ohlone
  \{
  */

+ #include "opentcp.h"
+
  //=========================== define ==========================================

  //=========================== typedef =========================================

! //=========================== module variables ================================
!
! typedef struct {
!    OpenQueueEntry_t*    pkt;
!    bool                 sending;
!    uint16_t             httpChunk;
!    uint8_t              getRequest[TCP_DEFAULT_WINDOW_SIZE];
! } ohlone_vars_t;

  //=========================== prototypes ======================================

! void ohlone_init(void);
! bool ohlone_shouldIlisten(void);
  void ohlone_receive(OpenQueueEntry_t* msg);
! void ohlone_sendDone(OpenQueueEntry_t* msg, owerror_t error);
! void ohlone_connectDone(owerror_t error);
! bool ohlone_debugPrint(void);

  /**
  \}
