*** stock_iot-lab_M3/openwsn/07-App/heli/heli.h	Thu Apr 24 11:01:37 2014
--- riot-openwsn-wip/openwsn/07-App/heli/heli.h	Thu Apr 24 16:53:29 2014
***************
*** 2,13 ****
  #define __HELI_H

  /**
! \addtogroup App
  \{
  \addtogroup Heli
  \{
  */

  //=========================== define ==========================================

  //=========================== typedef =========================================
--- 2,15 ----
  #define __HELI_H

  /**
! \addtogroup AppUdp
  \{
  \addtogroup Heli
  \{
  */

+
+ #include "openwsn.h"
  //=========================== define ==========================================

  //=========================== typedef =========================================
***************
*** 18,24 ****

  void heli_init();
  void heli_trigger();
! void heli_sendDone(OpenQueueEntry_t* msg, error_t error);
  void heli_receive(OpenQueueEntry_t* msg);
  bool heli_debugPrint();

--- 20,26 ----

  void heli_init();
  void heli_trigger();
! void heli_sendDone(OpenQueueEntry_t* msg, owerror_t error);
  void heli_receive(OpenQueueEntry_t* msg);
  bool heli_debugPrint();
