*** stock_iot-lab_M3/openwsn/07-App/tcpecho/tcpecho.h	Thu Apr 24 11:01:37 2014
--- riot-openwsn-wip/openwsn/07-App/tcpecho/tcpecho.h	Thu Apr 24 16:53:29 2014
***************
*** 2,8 ****
  #define __TCPECHO_H

  /**
! \addtogroup App
  \{
  \addtogroup tcpEcho
  \{
--- 2,8 ----
  #define __TCPECHO_H

  /**
! \addtogroup AppTcp
  \{
  \addtogroup tcpEcho
  \{
***************
*** 16,27 ****

  //=========================== prototypes ======================================

! void tcpecho_init();
! bool tcpecho_shouldIlisten();
  void tcpecho_receive(OpenQueueEntry_t* msg);
! void tcpecho_sendDone(OpenQueueEntry_t* msg, error_t error);
! void tcpecho_connectDone();
! bool tcpecho_debugPrint();

  /**
  \}
--- 16,27 ----

  //=========================== prototypes ======================================

! void tcpecho_init(void);
! bool tcpecho_shouldIlisten(void);
  void tcpecho_receive(OpenQueueEntry_t* msg);
! void tcpecho_sendDone(OpenQueueEntry_t* msg, owerror_t error);
! void tcpecho_connectDone(owerror_t error);
! bool tcpecho_debugPrint(void);

  /**
  \}
