*** stock_iot-lab_M3/openwsn/07-App/rheli/rheli.c	Thu Apr 24 11:01:37 2014
--- riot-openwsn-wip/openwsn/07-App/rheli/rheli.c	Thu Apr 24 16:55:54 2014
***************
*** 19,30 ****

  //=========================== prototypes ======================================

! error_t rheli_receive(OpenQueueEntry_t* msg,
                        coap_header_iht*  coap_header,
                        coap_option_iht*  coap_options);
  void    rheli_timer();
  void    rheli_sendDone(OpenQueueEntry_t* msg,
!                        error_t error);

  //=========================== public ==========================================

--- 19,30 ----

  //=========================== prototypes ======================================

! owerror_t rheli_receive(OpenQueueEntry_t* msg,
                        coap_header_iht*  coap_header,
                        coap_option_iht*  coap_options);
  void    rheli_timer();
  void    rheli_sendDone(OpenQueueEntry_t* msg,
!                        owerror_t error);

  //=========================== public ==========================================

***************
*** 49,58 ****

  //=========================== private =========================================

! error_t rheli_receive(OpenQueueEntry_t* msg,
                        coap_header_iht*  coap_header,
                        coap_option_iht*  coap_options) {
!    error_t outcome;

     if (coap_header->Code==COAP_CODE_REQ_POST) {

--- 49,58 ----

  //=========================== private =========================================

! owerror_t rheli_receive(OpenQueueEntry_t* msg,
                        coap_header_iht*  coap_header,
                        coap_option_iht*  coap_options) {
!    owerror_t outcome;

     if (coap_header->Code==COAP_CODE_REQ_POST) {

***************
*** 64,70 ****
        msg->length                      = 0;

        // set the CoAP header
-       coap_header->OC                  = 0;
        coap_header->Code                = COAP_CODE_RESP_CHANGED;

        outcome                          = E_SUCCESS;
--- 64,69 ----
***************
*** 79,84 ****
     heli_off();
  }

! void rheli_sendDone(OpenQueueEntry_t* msg, error_t error) {
     openqueue_freePacketBuffer(msg);
  }
\ No newline at end of file
--- 78,83 ----
     heli_off();
  }

! void rheli_sendDone(OpenQueueEntry_t* msg, owerror_t error) {
     openqueue_freePacketBuffer(msg);
  }
\ No newline at end of file
