*** stock_iot-lab_M3/openwsn/07-App/rleds/rleds.h	Thu Apr 24 11:01:37 2014
--- riot-openwsn-wip/openwsn/07-App/rleds/rleds.h	Thu Apr 24 16:53:29 2014
***************
*** 2,8 ****
  #define __RLEDS_H

  /**
! \addtogroup App
  \{
  \addtogroup netLeds
  \{
--- 2,8 ----
  #define __RLEDS_H

  /**
! \addtogroup AppCoAP
  \{
  \addtogroup netLeds
  \{
***************
*** 16,22 ****

  //=========================== prototypes ======================================

! void rleds_init();

  /**
  \}
--- 16,22 ----

  //=========================== prototypes ======================================

! void rleds__init();

  /**
  \}
