*** stock_iot-lab_M3/openwsn/07-App/tcpecho/tcpecho.c	Thu Apr 24 11:01:37 2014
--- riot-openwsn-wip/openwsn/07-App/tcpecho/tcpecho.c	Thu Apr 24 16:55:54 2014
***************
*** 10,19 ****

  //=========================== public ==========================================

! void tcpecho_init() {
  }

! bool tcpecho_shouldIlisten() {
     return TRUE;
  }

--- 10,19 ----

  //=========================== public ==========================================

! void tcpecho_init(void) {
  }

! bool tcpecho_shouldIlisten(void) {
     return TRUE;
  }

***************
*** 31,37 ****
     }
  }

! void tcpecho_sendDone(OpenQueueEntry_t* msg, error_t error) {
     msg->owner = COMPONENT_TCPECHO;
     if (msg->creator!=COMPONENT_TCPECHO) {
        openserial_printError(COMPONENT_TCPECHO,ERR_UNEXPECTED_SENDDONE,
--- 31,37 ----
     }
  }

! void tcpecho_sendDone(OpenQueueEntry_t* msg, owerror_t error) {
     msg->owner = COMPONENT_TCPECHO;
     if (msg->creator!=COMPONENT_TCPECHO) {
        openserial_printError(COMPONENT_TCPECHO,ERR_UNEXPECTED_SENDDONE,
***************
*** 43,52 ****
     openqueue_freePacketBuffer(msg);
  }

! void tcpecho_connectDone() {
  }

! bool tcpecho_debugPrint() {
     return FALSE;
  }

--- 43,52 ----
     openqueue_freePacketBuffer(msg);
  }

! void tcpecho_connectDone(owerror_t error) {
  }

! bool tcpecho_debugPrint(void) {
     return FALSE;
  }
