*** stock_iot-lab_M3/openwsn/07-App/imu/imu.c	Thu Apr 24 11:01:37 2014
--- riot-openwsn-wip/openwsn/07-App/imu/imu.c	Thu Apr 24 16:53:29 2014
***************
*** 54,60 ****
  }

  //I just sent a IMU packet, check I need to resend one
! void imu_sendDone(OpenQueueEntry_t* msg, error_t error) {
     msg->owner = COMPONENT_IMU;
     if (msg->creator!=COMPONENT_IMU) {
        openserial_printError(COMPONENT_IMU,ERR_UNEXPECTED_SENDDONE,
--- 54,60 ----
  }

  //I just sent a IMU packet, check I need to resend one
! void imu_sendDone(OpenQueueEntry_t* msg, owerror_t error) {
     msg->owner = COMPONENT_IMU;
     if (msg->creator!=COMPONENT_IMU) {
        openserial_printError(COMPONENT_IMU,ERR_UNEXPECTED_SENDDONE,
