*** stock_iot-lab_M3/openwsn/07-App/udpprint/udpprint.c	Thu Apr 24 11:01:37 2014
--- riot-openwsn-wip/openwsn/07-App/udpprint/udpprint.c	Thu Apr 24 16:55:54 2014
***************
*** 9,18 ****

  //=========================== public ==========================================

! void udpprint_init() {
  }

! void udpprint_sendDone(OpenQueueEntry_t* msg, error_t error) {
     openserial_printError(COMPONENT_UDPPRINT,ERR_UNEXPECTED_SENDDONE,
                           (errorparameter_t)0,
                           (errorparameter_t)0);
--- 9,18 ----

  //=========================== public ==========================================

! void udpprint_init(void) {
  }

! void udpprint_sendDone(OpenQueueEntry_t* msg, owerror_t error) {
     openserial_printError(COMPONENT_UDPPRINT,ERR_UNEXPECTED_SENDDONE,
                           (errorparameter_t)0,
                           (errorparameter_t)0);
***************
*** 24,30 ****
     openqueue_freePacketBuffer(msg);
  }

! bool udpprint_debugPrint() {
     return FALSE;
  }

--- 24,30 ----
     openqueue_freePacketBuffer(msg);
  }

! bool udpprint_debugPrint(void) {
     return FALSE;
  }
