*** stock_iot-lab_M3/openwsn/07-App/udpprint/udpprint.h	Thu Apr 24 11:01:37 2014
--- riot-openwsn-wip/openwsn/07-App/udpprint/udpprint.h	Thu Apr 24 16:53:30 2014
***************
*** 2,8 ****
  #define __UDPPRINT_H

  /**
! \addtogroup App
  \{
  \addtogroup udpPrint
  \{
--- 2,8 ----
  #define __UDPPRINT_H

  /**
! \addtogroup AppUdp
  \{
  \addtogroup udpPrint
  \{
***************
*** 16,25 ****

  //=========================== prototypes ======================================

! void udpprint_init();
! void udpprint_sendDone(OpenQueueEntry_t* msg, error_t error);
  void udpprint_receive(OpenQueueEntry_t* msg);
! bool udpprint_debugPrint();

  /**
  \}
--- 16,25 ----

  //=========================== prototypes ======================================

! void udpprint_init(void);
! void udpprint_sendDone(OpenQueueEntry_t* msg, owerror_t error);
  void udpprint_receive(OpenQueueEntry_t* msg);
! bool udpprint_debugPrint(void);

  /**
  \}
