*** stock_iot-lab_M3/openwsn/07-App/ohlone/ohlone_webpages.c	Thu Apr 24 11:01:37 2014
--- riot-openwsn-wip/openwsn/07-App/ohlone/ohlone_webpages.c	Thu Apr 24 16:55:54 2014
***************
*** 25,31 ****

  //=========================== public ==========================================

! void ohlone_webpages_init() {
     /*
     if (*(&eui64+3)==0x09) {                      // this is a GINA board (not a basestation)
        gyro_init();
--- 25,31 ----

  //=========================== public ==========================================

! void ohlone_webpages_init(void) {
     /*
     if (*(&eui64+3)==0x09) {                      // this is a GINA board (not a basestation)
        gyro_init();
