*** stock_iot-lab_M3/openwsn/07-App/tcpinject/tcpinject.h	Thu Apr 24 11:01:37 2014
--- riot-openwsn-wip/openwsn/07-App/tcpinject/tcpinject.h	Thu Apr 24 16:53:30 2014
***************
*** 2,8 ****
  #define __TCPINJECT_H

  /**
! \addtogroup App
  \{
  \addtogroup tcpInject
  \{
--- 2,8 ----
  #define __TCPINJECT_H

  /**
! \addtogroup AppTcp
  \{
  \addtogroup tcpInject
  \{
***************
*** 12,28 ****

  //=========================== typedef =========================================

! //=========================== variables =======================================

  //=========================== prototypes ======================================

! void tcpinject_init();
! bool tcpinject_shouldIlisten();
! void tcpinject_trigger();
! void tcpinject_connectDone(error_t error);
! void tcpinject_sendDone(OpenQueueEntry_t* msg, error_t error);
  void tcpinject_receive(OpenQueueEntry_t* msg);
! bool tcpinject_debugPrint();

  /**
  \}
--- 12,35 ----

  //=========================== typedef =========================================

! //=========================== module variables ================================
!
! typedef struct {
!    OpenQueueEntry_t*    pkt;
!    bool                 sending;
!    open_addr_t          hisAddress;
!    uint16_t             hisPort;
! } tcpinject_vars_t;

  //=========================== prototypes ======================================

! void tcpinject_init(void);
! bool tcpinject_shouldIlisten(void);
! void tcpinject_trigger(void);
! void tcpinject_connectDone(owerror_t error);
! void tcpinject_sendDone(OpenQueueEntry_t* msg, owerror_t error);
  void tcpinject_receive(OpenQueueEntry_t* msg);
! bool tcpinject_debugPrint(void);

  /**
  \}
