*** stock_iot-lab_M3/openwsn/07-App/udplatency/udplatency.h	Thu Apr 24 11:01:37 2014
--- riot-openwsn-wip/openwsn/07-App/udplatency/udplatency.h	Thu Apr 24 16:53:30 2014
***************
*** 2,27 ****
  #define __UDPLATENCY_H

  /**
! \addtogroup App
!
! \addtogroup udpLatency
  \{
  */

  //=========================== define ==========================================

  //=========================== typedef =========================================

  //=========================== variables =======================================

  //=========================== prototypes ======================================

! void udplatency_init();
! void udplatency_trigger();
! void udplatency_sendDone(OpenQueueEntry_t* msg, error_t error);
  void udplatency_receive(OpenQueueEntry_t* msg);
! bool udplatency_debugPrint();
! void udplatency_task();

  /**
  \}
--- 2,31 ----
  #define __UDPLATENCY_H

  /**
! \addtogroup AppUdp
! \{
! \addtogroup UdpLatency
  \{
  */

  //=========================== define ==========================================

+ /// inter-packet period (in mseconds)
+ #define UDPLATENCYPERIOD     3000
+ #define NUMPKTTEST           300
+
  //=========================== typedef =========================================

  //=========================== variables =======================================

  //=========================== prototypes ======================================

! void udplatency_init(void);
! void udplatency_trigger(void);
! void udplatency_sendDone(OpenQueueEntry_t* msg, owerror_t error);
  void udplatency_receive(OpenQueueEntry_t* msg);
! bool udplatency_debugPrint(void);
! void udplatency_task(void);

  /**
  \}
