*** stock_iot-lab_M3/openwsn/07-App/ohlone/ohlone_webpages.h	Thu Apr 24 11:01:37 2014
--- riot-openwsn-wip/openwsn/07-App/ohlone/ohlone_webpages.h	Thu Apr 24 16:55:54 2014
***************
*** 27,33 ****

  //=========================== prototypes ======================================

! void     ohlone_webpages_init();
  uint8_t  ohlone_webpage(uint8_t *getRequest, uint16_t chunk, uint8_t *packet);

  #endif
\ No newline at end of file
--- 27,33 ----

  //=========================== prototypes ======================================

! void     ohlone_webpages_init(void);
  uint8_t  ohlone_webpage(uint8_t *getRequest, uint16_t chunk, uint8_t *packet);

  #endif
\ No newline at end of file
