*** stock_iot-lab_M3/openwsn/07-App/udpecho/udpecho.c	Thu Apr 24 11:01:37 2014
--- riot-openwsn-wip/openwsn/07-App/udpecho/udpecho.c	Thu Apr 24 16:55:54 2014
***************
*** 10,16 ****

  //=========================== public ==========================================

! void udpecho_init() {
  }

  void udpecho_receive(OpenQueueEntry_t* msg) {
--- 10,16 ----

  //=========================== public ==========================================

! void udpecho_init(void) {
  }

  void udpecho_receive(OpenQueueEntry_t* msg) {
***************
*** 44,50 ****
     }
  }

! void udpecho_sendDone(OpenQueueEntry_t* msg, error_t error) {
     msg->owner = COMPONENT_UDPECHO;
     if (msg->creator!=COMPONENT_UDPECHO) {
        openserial_printError(COMPONENT_UDPECHO,ERR_UNEXPECTED_SENDDONE,
--- 44,50 ----
     }
  }

! void udpecho_sendDone(OpenQueueEntry_t* msg, owerror_t error) {
     msg->owner = COMPONENT_UDPECHO;
     if (msg->creator!=COMPONENT_UDPECHO) {
        openserial_printError(COMPONENT_UDPECHO,ERR_UNEXPECTED_SENDDONE,
***************
*** 54,60 ****
     openqueue_freePacketBuffer(msg);
  }

! bool udpecho_debugPrint() {
     return FALSE;
  }

--- 54,60 ----
     openqueue_freePacketBuffer(msg);
  }

! bool udpecho_debugPrint(void) {
     return FALSE;
  }
