*** stock_iot-lab_M3/openwsn/07-App/udpecho/udpecho.h	Thu Apr 24 11:01:37 2014
--- riot-openwsn-wip/openwsn/07-App/udpecho/udpecho.h	Thu Apr 24 16:53:30 2014
***************
*** 2,8 ****
  #define __UDPECHO_H

  /**
! \addtogroup App
  \{
  \addtogroup udpEcho
  \{
--- 2,8 ----
  #define __UDPECHO_H

  /**
! \addtogroup AppUdp
  \{
  \addtogroup udpEcho
  \{
***************
*** 16,25 ****

  //=========================== prototypes ======================================

! void udpecho_init();
  void udpecho_receive(OpenQueueEntry_t* msg);
! void udpecho_sendDone(OpenQueueEntry_t* msg, error_t error);
! bool udpecho_debugPrint();

  /**
  \}
--- 16,25 ----

  //=========================== prototypes ======================================

! void udpecho_init(void);
  void udpecho_receive(OpenQueueEntry_t* msg);
! void udpecho_sendDone(OpenQueueEntry_t* msg, owerror_t error);
! bool udpecho_debugPrint(void);

  /**
  \}
