*** stock_iot-lab_M3/openwsn/07-App/rinfo/rinfo.h	Thu Apr 24 11:01:37 2014
--- riot-openwsn-wip/openwsn/07-App/rinfo/rinfo.h	Thu Apr 24 16:53:29 2014
***************
*** 2,8 ****
  #define __RINFO_H

  /**
! \addtogroup App
  \{
  \addtogroup rXL1
  \{
--- 2,8 ----
  #define __RINFO_H

  /**
! \addtogroup AppCoAP
  \{
  \addtogroup rXL1
  \{
***************
*** 16,22 ****

  //=========================== prototypes ======================================

! void rinfo_init();

  /**
  \}
--- 16,22 ----

  //=========================== prototypes ======================================

! void rinfo_init(void);

  /**
  \}
