*** stock_iot-lab_M3/openwsn/07-App/udpinject/udpinject.c	Thu Apr 24 11:01:37 2014
--- riot-openwsn-wip/openwsn/07-App/udpinject/udpinject.c	Thu Apr 24 16:55:54 2014
***************
*** 11,20 ****

  //=========================== public ==========================================

! void udpinject_init() {
  }

! void udpinject_trigger() {
     OpenQueueEntry_t* pkt;
     uint8_t number_bytes_from_input_buffer;
     uint8_t input_buffer[18];
--- 11,20 ----

  //=========================== public ==========================================

! void udpinject_init(void) {
  }

! void udpinject_trigger(void) {
     OpenQueueEntry_t* pkt;
     uint8_t number_bytes_from_input_buffer;
     uint8_t input_buffer[18];
***************
*** 54,60 ****
     }
  }

! void udpinject_sendDone(OpenQueueEntry_t* msg, error_t error) {
     msg->owner = COMPONENT_UDPINJECT;
     if (msg->creator!=COMPONENT_UDPINJECT) {
        openserial_printError(COMPONENT_UDPINJECT,ERR_UNEXPECTED_SENDDONE,
--- 54,60 ----
     }
  }

! void udpinject_sendDone(OpenQueueEntry_t* msg, owerror_t error) {
     msg->owner = COMPONENT_UDPINJECT;
     if (msg->creator!=COMPONENT_UDPINJECT) {
        openserial_printError(COMPONENT_UDPINJECT,ERR_UNEXPECTED_SENDDONE,
***************
*** 68,74 ****
     openqueue_freePacketBuffer(msg);
  }

! bool udpinject_debugPrint() {
     return FALSE;
  }

--- 68,74 ----
     openqueue_freePacketBuffer(msg);
  }

! bool udpinject_debugPrint(void) {
     return FALSE;
  }
