*** stock_iot-lab_M3/openwsn/board_ow.h	Thu Apr 24 11:01:37 2014
--- riot-openwsn-wip/openwsn/board_ow.h	Thu Apr 24 16:53:30 2014
***************
*** 1,12 ****
  /**
  \brief Cross-platform declaration "board" bsp module.

  \author Thomas Watteyne <watteyne@eecs.berkeley.edu>, February 2012.
  */

- #ifndef __BOARD_H
- #define __BOARD_H
-
  #include "board_info.h"

  //=========================== define ==========================================
--- 1,17 ----
+ #ifndef __BOARD_H
+ #define __BOARD_H
+
  /**
+ \addtogroup BSP
+ \{
+ \addtogroup board
+ \{
+
  \brief Cross-platform declaration "board" bsp module.

  \author Thomas Watteyne <watteyne@eecs.berkeley.edu>, February 2012.
  */

  #include "board_info.h"

  //=========================== define ==========================================
***************
*** 16,29 ****
     KICK_SCHEDULER,
  } kick_scheduler_t;

  //=========================== typedef =========================================

  //=========================== variables =======================================

  //=========================== prototypes ======================================

! void board_init();
! void board_sleep();
! void board_reset();

  #endif
--- 21,49 ----
     KICK_SCHEDULER,
  } kick_scheduler_t;

+ #if defined(__GNUC__) && (__GNUC__==4)  && (__GNUC_MINOR__<=5) && defined(__MSP430__)
+    // mspgcc <4.5.x
+ #include <signal.h>
+ #define ISR(v) interrupt (v ## _VECTOR) v ## _ISR(void)
+ #else
+    // other
+ #define __PRAGMA__(x) _Pragma(#x)
+ #define ISR(v) __PRAGMA__(vector=v ##_VECTOR) __interrupt void v ##_ISR(void)
+ #endif
+
  //=========================== typedef =========================================

  //=========================== variables =======================================

  //=========================== prototypes ======================================

! void board_init_ow(void);
! void board_sleep(void);
! void board_reset(void);
!
! /**
! \}
! \}
! */

  #endif
