*** stock_iot-lab_M3/openwsn/cross-layers/packetfunctions.h	Thu Apr 24 11:01:37 2014
--- riot-openwsn-wip/openwsn/cross-layers/packetfunctions.h	Thu Apr 24 16:55:54 2014
***************
*** 2,8 ****
  #define __PACKETFUNCTIONS_H

  /**
! \addtogroup helpers
  \{
  \addtogroup PacketFunctions
  \{
--- 2,8 ----
  #define __PACKETFUNCTIONS_H

  /**
! \addtogroup cross-layers
  \{
  \addtogroup PacketFunctions
  \{
