*** stock_iot-lab_M3/openwsn/cross-layers/idmanager.h	Thu Apr 24 11:01:37 2014
--- riot-openwsn-wip/openwsn/cross-layers/idmanager.h	Thu Apr 24 16:53:30 2014
***************
*** 12,17 ****
--- 12,21 ----

  //=========================== define ==========================================

+ #define ACTION_YES      'Y'
+ #define ACTION_NO       'N'
+ #define ACTION_TOGGLE   'T'
+
  //=========================== typedef =========================================

  typedef struct {
***************
*** 23,44 ****
     open_addr_t   myPrefix;
  } debugIDManagerEntry_t;

! //=========================== variables =======================================

  //=========================== prototypes ======================================

! void         idmanager_init();
! bool         idmanager_getIsDAGroot();
  void         idmanager_setIsDAGroot(bool newRole);
! bool         idmanager_getIsBridge();
  void         idmanager_setIsBridge(bool newRole);
  open_addr_t* idmanager_getMyID(uint8_t type);
! error_t      idmanager_setMyID(open_addr_t* newID);
  bool         idmanager_isMyAddress(open_addr_t* addr);
! void         idmanager_triggerAboutRoot();
! void         idmanager_triggerAboutBridge();

! bool         debugPrint_id();


  /**
--- 27,57 ----
     open_addr_t   myPrefix;
  } debugIDManagerEntry_t;

! //=========================== module variables ================================
!
! typedef struct {
!    bool          isDAGroot;
!    bool          isBridge;
!    open_addr_t   my16bID;
!    open_addr_t   my64bID;
!    open_addr_t   myPANID;
!    open_addr_t   myPrefix;
! } idmanager_vars_t;

  //=========================== prototypes ======================================

! void         idmanager_init(void);
! bool         idmanager_getIsDAGroot(void);
  void         idmanager_setIsDAGroot(bool newRole);
! bool         idmanager_getIsBridge(void);
  void         idmanager_setIsBridge(bool newRole);
  open_addr_t* idmanager_getMyID(uint8_t type);
! owerror_t      idmanager_setMyID(open_addr_t* newID);
  bool         idmanager_isMyAddress(open_addr_t* addr);
! void         idmanager_triggerAboutRoot(void);
! void         idmanager_triggerAboutBridge(void);

! bool         debugPrint_id(void);


  /**
