*** stock_iot-lab_M3/openwsn/cross-layers/openqueue.h	Thu Apr 24 11:01:37 2014
--- riot-openwsn-wip/openwsn/cross-layers/openqueue.h	Thu Apr 24 16:53:30 2014
***************
*** 22,45 ****
     uint8_t  owner;
  } debugOpenQueueEntry_t;

! //=========================== variables =======================================

  //=========================== prototypes ======================================

  // admin
! void               openqueue_init();
! bool               debugPrint_queue();
  // called by any component
  OpenQueueEntry_t*  openqueue_getFreePacketBuffer(uint8_t creator);
! error_t            openqueue_freePacketBuffer(OpenQueueEntry_t* pkt);
  void               openqueue_removeAllCreatedBy(uint8_t creator);
  void               openqueue_removeAllOwnedBy(uint8_t owner);
  // called by res
! OpenQueueEntry_t*  openqueue_resGetSentPacket();
! OpenQueueEntry_t*  openqueue_resGetReceivedPacket();
  // called by IEEE80215E
  OpenQueueEntry_t*  openqueue_macGetDataPacket(open_addr_t* toNeighbor);
! OpenQueueEntry_t*  openqueue_macGetAdvPacket();

  /**
  \}
--- 22,49 ----
     uint8_t  owner;
  } debugOpenQueueEntry_t;

! //=========================== module variables ================================
!
! typedef struct {
!    OpenQueueEntry_t queue[QUEUELENGTH];
! } openqueue_vars_t;

  //=========================== prototypes ======================================

  // admin
! void               openqueue_init(void);
! bool               debugPrint_queue(void);
  // called by any component
  OpenQueueEntry_t*  openqueue_getFreePacketBuffer(uint8_t creator);
! owerror_t         openqueue_freePacketBuffer(OpenQueueEntry_t* pkt);
  void               openqueue_removeAllCreatedBy(uint8_t creator);
  void               openqueue_removeAllOwnedBy(uint8_t owner);
  // called by res
! OpenQueueEntry_t*  openqueue_resGetSentPacket(void);
! OpenQueueEntry_t*  openqueue_resGetReceivedPacket(void);
  // called by IEEE80215E
  OpenQueueEntry_t*  openqueue_macGetDataPacket(open_addr_t* toNeighbor);
! OpenQueueEntry_t*  openqueue_macGetAdvPacket(void);

  /**
  \}
