*** stock_iot-lab_M3/openwsn/debugpins.c	Thu Apr 24 11:19:39 2014
--- riot-openwsn-wip/openwsn/debugpins.c	Thu Apr 24 16:53:30 2014
***************
*** 0 ****
--- 1,42 ----
+ /**
+ \brief iot-lab_M3 definition of the "debugpins" bsp module.
+
+ \author Alaeddine Weslati <alaeddine.weslati@inria.fr>,  January 2014.
+ */
+ #include "stm32f10x_lib.h"
+ #include "debugpins.h"
+
+ //=========================== defines =========================================
+
+ //=========================== variables =======================================
+
+ //=========================== prototypes ======================================
+
+ //=========================== public ==========================================
+
+ void debugpins_init() {}
+
+ void debugpins_frame_toggle() {}
+ void debugpins_frame_clr() {}
+ void debugpins_frame_set() {}
+
+ void debugpins_slot_toggle() {}
+ void debugpins_slot_clr() {}
+ void debugpins_slot_set() {}
+
+ void debugpins_fsm_toggle() {}
+ void debugpins_fsm_clr() {}
+ void debugpins_fsm_set() {}
+
+ void debugpins_task_toggle() {}
+ void debugpins_task_clr() {}
+ void debugpins_task_set() {}
+
+ void debugpins_isr_toggle() {}
+ void debugpins_isr_clr() {}
+ void debugpins_isr_set() {}
+
+ void debugpins_radio_toggle() {}
+ void debugpins_radio_clr() {}
+ void debugpins_radio_set() {}
+
