*** stock_iot-lab_M3/openwsn/eui64.h	Thu Apr 24 11:01:37 2014
--- riot-openwsn-wip/openwsn/eui64.h	Thu Apr 24 16:55:54 2014
***************
*** 1,12 ****
  /**
  \brief Cross-platform declaration "eui64" bsp module.

  \author Thomas Watteyne <watteyne@eecs.berkeley.edu>, March 2012.
  */

- #ifndef __EUI64_H
- #define __EUI64_H
-
  #include <stdint.h>

  //=========================== define ==========================================
--- 1,17 ----
+ #ifndef __EUI64_H
+ #define __EUI64_H
+
  /**
+ \addtogroup BSP
+ \{
+ \addtogroup eui64
+ \{
+
  \brief Cross-platform declaration "eui64" bsp module.

  \author Thomas Watteyne <watteyne@eecs.berkeley.edu>, March 2012.
  */

  #include <stdint.h>

  //=========================== define ==========================================
***************
*** 19,22 ****
--- 24,32 ----

  void eui64_get(uint8_t* addressToWrite);

+ /**
+ \}
+ \}
+ */
+
  #endif
\ No newline at end of file
