*** stock_iot-lab_M3/openwsn/exti.c	Thu Apr 24 11:19:39 2014
--- riot-openwsn-wip/openwsn/exti.c	Thu Apr 24 16:53:30 2014
***************
*** 0 ****
--- 1,14 ----
+ /**
+ \brief iot-lab_M3 definition of the "exti" bsp module.
+
+ \author Alaeddine Weslati <alaeddine.weslati@inria.fr>, January 2014.
+ */
+
+ // Includes ------------------------------------------------------------------*/
+ #include "stm32f10x_lib.h"
+ #include "stm32f10x_gpio.h"
+
+ void Exti_Init(void)
+ {
+ }
+
