*** stock_iot-lab_M3/openwsn/gpio.h	Thu Apr 24 11:19:39 2014
--- riot-openwsn-wip/openwsn/gpio.h	Thu Apr 24 16:53:30 2014
***************
*** 0 ****
--- 1,13 ----
+ #ifndef __GPIO_H
+ #define __GPIO_H
+
+ //=========================== defines =========================================
+
+ //=========================== variables =======================================
+
+ //=========================== prototypes ======================================
+
+ void GPIO_Config_ALL_AIN(void);
+ void GPIO_Configuration(void);
+
+ #endif
