*** stock_iot-lab_M3/openwsn/leds_ow.h	Thu Apr 24 11:01:37 2014
--- riot-openwsn-wip/openwsn/leds_ow.h	Thu Apr 24 16:55:54 2014
***************
*** 1,12 ****
  /**
  \brief Cross-platform declaration "leds" bsp module.

  \author Thomas Watteyne <watteyne@eecs.berkeley.edu>, February 2012.
  */

- #ifndef __LEDS_H
- #define __LEDS_H
-
  #include "stdint.h"

  //=========================== define ==========================================
--- 1,17 ----
+ #ifndef __LEDS_H
+ #define __LEDS_H
+
  /**
+ \addtogroup BSP
+ \{
+ \addtogroup leds
+ \{
+
  \brief Cross-platform declaration "leds" bsp module.

  \author Thomas Watteyne <watteyne@eecs.berkeley.edu>, February 2012.
  */

  #include "stdint.h"

  //=========================== define ==========================================
***************
*** 17,50 ****

  //=========================== prototypes ======================================

! void    leds_init();

! void    leds_error_on();
! void    leds_error_off();
! void    leds_error_toggle();
! uint8_t leds_error_isOn();
! void    leds_error_blink();
!
! void    leds_radio_on();
! void    leds_radio_off();
! void    leds_radio_toggle();
! uint8_t leds_radio_isOn();
!
! void    leds_sync_on();
! void    leds_sync_off();
! void    leds_sync_toggle();
! uint8_t leds_sync_isOn();
!
! void    leds_debug_on();
! void    leds_debug_off();
! void    leds_debug_toggle();
! uint8_t leds_debug_isOn();
!
! void    leds_all_on();
! void    leds_all_off();
! void    leds_all_toggle();

! void    leds_circular_shift();
! void    leds_increment();

  #endif
--- 22,60 ----

  //=========================== prototypes ======================================

! void    leds_init_ow(void);
!
! void    leds_error_on(void);
! void    leds_error_off(void);
! void    leds_error_toggle(void);
! uint8_t leds_error_isOn(void);
! void    leds_error_blink(void);
!
! void    leds_radio_on(void);
! void    leds_radio_off(void);
! void    leds_radio_toggle(void);
! uint8_t leds_radio_isOn(void);
!
! void    leds_sync_on(void);
! void    leds_sync_off(void);
! void    leds_sync_toggle(void);
! uint8_t leds_sync_isOn(void);
!
! void    leds_debug_on(void);
! void    leds_debug_off(void);
! void    leds_debug_toggle(void);
! uint8_t leds_debug_isOn(void);
!
! void    leds_all_on(void);
! void    leds_all_off(void);
! void    leds_all_toggle(void);

! void    leds_circular_shift(void);
! void    leds_increment(void);

! /**
! \}
! \}
! */

  #endif
