*** stock_iot-lab_M3/openwsn/rcc.h	Thu Apr 24 11:19:40 2014
--- riot-openwsn-wip/openwsn/rcc.h	Thu Apr 24 16:55:54 2014
***************
*** 0 ****
--- 1,20 ----
+ /**
+ \brief openmoteSTM32 definition of the RCC.
+
+ \author Chang Tengfei <tengfei.chang@gmail.com>,  July 2012.
+ */
+
+ #ifndef __RCC_H
+ #define __RCC_H
+
+ //=========================== defines =========================================
+
+ //=========================== variables =======================================
+
+ //=========================== prototypes ======================================
+
+ void RCC_Configuration(void);
+
+ void RCC_Wakeup(void);
+
+ #endif
\ No newline at end of file
