*** stock_iot-lab_M3/openwsn/scheduler.h	Thu Apr 24 11:01:37 2014
--- riot-openwsn-wip/openwsn/scheduler.h	Thu Apr 24 16:53:30 2014
***************
*** 2,8 ****
  #define __SCHEDULER_H

  /**
! \addtogroup drivers
  \{
  \addtogroup Scheduler
  \{
--- 2,8 ----
  #define __SCHEDULER_H

  /**
! \addtogroup kernel
  \{
  \addtogroup Scheduler
  \{
***************
*** 33,51 ****

  typedef void (*task_cbt)(void);

! //=========================== variables =======================================

  //=========================== prototypes ======================================

  // public functions
! void scheduler_init();
! void scheduler_start();
  void scheduler_push_task(task_cbt task_cb, task_prio_t prio);

  // interrupt handlers
! void isr_ieee154e_newSlot();
! void isr_ieee154e_timer();
! void isr_adc();
  #ifdef ISR_GYRO
  void isr_gyro();
  #endif
--- 33,69 ----

  typedef void (*task_cbt)(void);

! typedef struct task_llist_t {
!    task_cbt             cb;
!    task_prio_t          prio;
!    void*                next;
! } taskList_item_t;
!
! //=========================== module variables ================================
!
! typedef struct {
!    taskList_item_t      taskBuf[TASK_LIST_DEPTH];
!    taskList_item_t*     task_list;
!    uint8_t              numTasksCur;
!    uint8_t              numTasksMax;
! } scheduler_vars_t;
!
! typedef struct {
!    uint8_t              numTasksCur;
!    uint8_t              numTasksMax;
! } scheduler_dbg_t;

  //=========================== prototypes ======================================

  // public functions
! void scheduler_init(void);
! void scheduler_start(void);
  void scheduler_push_task(task_cbt task_cb, task_prio_t prio);

  // interrupt handlers
! void isr_ieee154e_newSlot(void);
! void isr_ieee154e_timer(void);
! void isr_adc(void);
  #ifdef ISR_GYRO
  void isr_gyro();
  #endif
