*** stock_iot-lab_M3/openwsn/spi.h	Thu Apr 24 11:01:37 2014
--- riot-openwsn-wip/openwsn/spi.h	Thu Apr 24 16:53:30 2014
***************
*** 8,14 ****
  #define __SPI_H

  #include "stdint.h"
! #include "board.h"

  //=========================== define ==========================================

--- 8,14 ----
  #define __SPI_H

  #include "stdint.h"
! #include "board_ow.h"

  //=========================== define ==========================================

***************
*** 49,55 ****

  //=========================== prototypes ======================================

! void    spi_init();
  #ifdef SPI_IN_INTERRUPT_MODE
  void    spi_setCb(spi_cbt cb);
  #endif
--- 49,55 ----

  //=========================== prototypes ======================================

! void    spi_init(void);
  #ifdef SPI_IN_INTERRUPT_MODE
  void    spi_setCb(spi_cbt cb);
  #endif
***************
*** 62,67 ****
                   spi_last_t   isLast);

  // interrupt handlers
! kick_scheduler_t spi_isr();

  #endif
--- 62,67 ----
                   spi_last_t   isLast);

  // interrupt handlers
! kick_scheduler_t spi_isr(void);

  #endif
