From e51bbd7efb1d01f052efef756a0a7423d7b8078a Mon Sep 17 00:00:00 2001
From: Frederik Haxel <haxel@fzi.de>
Date: Mon, 14 Aug 2023 17:26:11 +0200
Subject: [PATCH] 64bit fix

---
 src/fdb_kvdb.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/fdb_kvdb.c b/src/fdb_kvdb.c
index 393c463..7905fda 100644
--- a/src/fdb_kvdb.c
+++ b/src/fdb_kvdb.c
@@ -341,7 +341,7 @@ static fdb_err_t read_kv(fdb_kvdb_t db, fdb_kv_t kv)
     kv->status = (fdb_kv_status_t) _fdb_get_status(kv_hdr.status_table, FDB_KV_STATUS_NUM);
     kv->len = kv_hdr.len;
 
-    if (kv->len == ~0UL || kv->len > db_max_size(db) || kv->len < KV_HDR_DATA_SIZE) {
+    if (kv->len == UINT32_MAX || kv->len > db_max_size(db) || kv->len < KV_HDR_DATA_SIZE) {
         /* the KV length was not write, so reserved the info for current KV */
         kv->len = KV_HDR_DATA_SIZE;
         if (kv->status != FDB_KV_ERR_HDR) {
-- 
2.34.1

