From 47b9fba23bedff6e28576271291a6ca7958a3caa Mon Sep 17 00:00:00 2001
From: Alexandre Abadie <alexandre.abadie@inria.fr>
Date: Mon, 8 Jan 2018 10:45:28 +0100
Subject: [PATCH] accept all join retries values in EU868

---
 src/mac/region/RegionEU868.c | 7 ++-----
 1 file changed, 2 insertions(+), 5 deletions(-)

diff --git a/src/mac/region/RegionEU868.c b/src/mac/region/RegionEU868.c
index 3a27f42..e7e6f54 100644
--- a/src/mac/region/RegionEU868.c
+++ b/src/mac/region/RegionEU868.c
@@ -393,11 +393,8 @@ bool RegionEU868Verify( VerifyParams_t* verify, PhyAttribute_t phyAttribute )
         }
         case PHY_NB_JOIN_TRIALS:
         {
-            if( verify->NbJoinTrials < 48 )
-            {
-                return false;
-            }
-            break;
+            /* accept all values */
+            return true;
         }
         default:
             return false;
-- 
2.14.1

