From 8af8b1d8b148acf089df1f5dbb794bd6f4c564dd Mon Sep 17 00:00:00 2001
From: Marian Buschsieweke <marian.buschsieweke@ovgu.de>
Date: Thu, 8 Sep 2022 20:40:02 +0200
Subject: [PATCH] src/PosixSerialPort.cpp: Add missing include

---
 src/PosixSerialPort.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/PosixSerialPort.cpp b/src/PosixSerialPort.cpp
index 4ca7954..b877e29 100644
--- a/src/PosixSerialPort.cpp
+++ b/src/PosixSerialPort.cpp
@@ -36,6 +36,7 @@
 #include <termios.h>
 #include <errno.h>
 #include <sys/ioctl.h>
+#include <sys/select.h>
 
 #include <string>
 
-- 
2.37.1

