From 4284cdec1632034280d0d5764f1f1957f956e657 Mon Sep 17 00:00:00 2001
From: Hendrik van Essen <hendrik.ve@fu-berlin.de>
Date: Sat, 6 Feb 2021 00:01:55 +0100
Subject: [PATCH 8/8] multithread: enable overriding of TEST_MESSAGE

---
 wolfmqtt_examples/multithread/multithread.c | 12 ++++++++----
 1 file changed, 8 insertions(+), 4 deletions(-)

diff --git a/wolfmqtt_examples/multithread/multithread.c b/wolfmqtt_examples/multithread/multithread.c
index df84df4..d33078e 100755
--- a/wolfmqtt_examples/multithread/multithread.c
+++ b/wolfmqtt_examples/multithread/multithread.c
@@ -38,7 +38,11 @@
 /* Maximum size for network read/write callbacks. There is also a v5 define that
    describes the max MQTT control packet size, DEFAULT_MAX_PKT_SZ. */
 #define MAX_BUFFER_SIZE 1024
-#define TEST_MESSAGE    "test00"
+
+#ifndef TEST_MESSAGE
+#define TEST_MESSAGE    "(  ) test"
+#endif
+
 /* Number of publish tasks. Each will send a unique message to the broker. */
 #define NUM_PUB_TASKS   10
 
@@ -457,7 +461,7 @@ static void *publish_task(void *param)
 #endif
 {
     int rc;
-    char buf[7];
+    char buf[50];
     MQTTCtx *mqttCtx = (MQTTCtx*)param;
     MqttPublish publish;
 
@@ -469,8 +473,8 @@ static void *publish_task(void *param)
     publish.topic_name = mqttCtx->topic_name;
     publish.packet_id = mqtt_get_packetid_threadsafe();
     XSTRNCPY(buf, TEST_MESSAGE, sizeof(buf));
-    buf[4] = '0' + ((publish.packet_id / 10) % 10);
-    buf[5] = '0' + (publish.packet_id % 10);
+    buf[1] = '0' + ((publish.packet_id / 10) % 10);
+    buf[2] = '0' + (publish.packet_id % 10);
     publish.buffer = (byte*)buf;
     publish.total_len = (word16)XSTRLEN(buf);
 
-- 
2.25.1

