From 27bbb1a6d08e997fd3f21bffd654941a3269dea6 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Jos=C3=A9=20Ignacio=20Alamos=20Aste?= <jialamos@uc.cl>
Date: Thu, 14 Jul 2016 17:27:41 +0200
Subject: [PATCH] Fixed

---
 src/core/mac/mac.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/core/mac/mac.cpp b/src/core/mac/mac.cpp
index a5c52b8..addb111 100644
--- a/src/core/mac/mac.cpp
+++ b/src/core/mac/mac.cpp
@@ -742,7 +742,7 @@ ThreadError Mac::ProcessReceiveSecurity(Frame &aFrame, const Address &aSrcAddr,
     }
 
     VerifyOrExit(keySequence > aNeighbor->mKeySequence ||
-                 (keySequence == aNeighbor->mKeySequence) && (frameCounter >= aNeighbor->mValid.mLinkFrameCounter),
+            ((keySequence == aNeighbor->mKeySequence) && (frameCounter >= aNeighbor->mValid.mLinkFrameCounter)),
                  error = kThreadError_Security);
 
     aesCcm.SetKey(macKey, 16);
-- 
2.7.4 (Apple Git-66)

