From cc4910d3d748c2593ccc6e8d4f816c96bca51c89 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Jos=C3=A9=20Ignacio=20Alamos=20Aste?= <jialamos@uc.cl>
Date: Fri, 15 Jul 2016 09:56:56 +0200
Subject: [PATCH] Fixed aliasing rule

---
 src/cli/cli.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/cli/cli.cpp b/src/cli/cli.cpp
index 4e27775..b9042db 100644
--- a/src/cli/cli.cpp
+++ b/src/cli/cli.cpp
@@ -639,7 +639,7 @@ exit:
 
 void Interpreter::HandlePingTimer(void *aContext)
 {
-    *(uint32_t *)sEchoRequest = HostSwap32(Timer::GetNow());
+    *(char *)sEchoRequest = HostSwap32(Timer::GetNow());
     sIcmpEcho.SendEchoRequest(sSockAddr, sEchoRequest, sLength);
     sCount--;
 
-- 
2.7.4 (Apple Git-66)

